frontend/lib/src/components/graph.dart

```diff
@@ -19,138 +18,213 @@ class Graph extends StatelessWidget {
-        child: InkWell(
-          onTap: onTap,
-          child: SizedBox(
-            width: double.maxFinite,
-            height: 236,
-            child: Row(
-              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
-              children: [
-                SizedBox(
-                  width: 150,
-                  height: 150,
-                  child: StatefulBuilder(
-                    builder: (context, stateful) {
-                      return PieChart(
-                        PieChartData(
-                          pieTouchData: PieTouchData(
-                            touchCallback:
-                                (FlTouchEvent event, pieTouchResponse) {
-                              stateful(() {
-                                if (!event.isInterestedForInteractions ||
-                                    pieTouchResponse == null ||
-                                    pieTouchResponse.touchedSection == null) {
-                                  graphController.setFocusIndex = -1;
-
-                                  return;
-                                }
-                                graphController.setFocusIndex = pieTouchResponse
-                                    .touchedSection!.touchedSectionIndex;
-                              });
-                            },
-                          ),
-                          borderData: FlBorderData(
-                            show: false,
-                          ),
-                          sectionsSpace: 0,
-                          centerSpaceRadius: 30,
-                          sections: showingSections(),
-                        ),
-                      );
-                    },
+    return FutureBuilder(
+      future: spendingBreakdownController.getSpendingBreakdown(
+        accountId: accountId,
+      ),
+      builder: (context, snapshot) {
+        if (snapshot.connectionState != ConnectionState.done) {
+          return const CircularProgressIndicator();
+        }
+
+        if (spendingBreakdownController.spendingList.isEmpty) {
+          return Center(
+            child: Container(
+              decoration: const BoxDecoration(
+                borderRadius: BorderRadius.all(
+                  Radius.circular(8.0),
+                ),
+                color: Colors.white,
+                boxShadow: [
+                  BoxShadow(
+                    color: Color.fromRGBO(0, 0, 0, 0.25),
+                    blurRadius: 2,
+                    spreadRadius: 1,
+                    offset: Offset(0, 2),
+                  ),
+                ],
+              ),
+              padding: const EdgeInsets.all(20),
+              width: double.maxFinite,
+              height: 236,
+              child: InkWell(
+                onTap: onTap,
+                child: Align(
+                  alignment: Alignment.center,
+                  child: Text(
+                    'No Data Found',
```

Comment:
refactor into a separate component, No Data found and main content has similar layout

for No data found, Inkwell is not necessary